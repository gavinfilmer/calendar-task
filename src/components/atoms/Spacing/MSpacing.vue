<template functional>
  <div
    class="m-spacing"
    :class="{
      [`m-spacing--top-${props.top}`]: props.top !== '',
      [`m-spacing--right-${props.right}`]: props.right !== '',
      [`m-spacing--bottom-${props.bottom}`]: props.bottom !== '',
      [`m-spacing--left-${props.left}`]: props.left !== '',
      'm-spacing--inline': props.inline,
    }"
  >
    <slot />
  </div>
</template>

<script>
export default {
  name: 'm-spacing',
  props: {
    top: {
      type: String,
      required: false,
      default: '',
    },
    right: {
      type: String,
      required: false,
      default: '',
    },
    bottom: {
      type: String,
      required: false,
      default: '',
    },
    left: {
      type: String,
      required: false,
      default: '',
    },
    inline: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../../sass/framework/framework.scss';
$spacing-vertical: (
  xxxs: $pad-v-xxxs,
  xxs: $pad-v-xxs,
  xs: $pad-v-xs,
  sm: $pad-v-sm,
  md: $pad-v-md,
  lg: $pad-v-lg,
  xl: $pad-v-xl,
  xxl: $pad-v-xxl,
  xxxl: $pad-v-xxxl,
);

$spacing-horizontal: (
  xs: $pad-h-xs,
  sm: $pad-h-sm,
  md: $pad-h-md,
  lg: $pad-h-lg,
  xl: $pad-h-xl,
  xxl: $pad-h-xxl,
  xxxl: $pad-h-xxxl,
  xxxxl: $pad-h-xxxxl,
);

.m-spacing {
  &--top {
    @each $key, $value in $spacing-vertical {
      &-#{$key} {
        margin-top: $value;
      }
    }
  }

  &--right {
    @each $key, $value in $spacing-horizontal {
      &-#{$key} {
        margin-right: $value;
      }
    }
  }

  &--bottom {
    @each $key, $value in $spacing-vertical {
      &-#{$key} {
        margin-bottom: $value;
      }
    }
  }

  &--left {
    @each $key, $value in $spacing-horizontal {
      &-#{$key} {
        margin-left: $value;
      }
    }
  }

  &--inline {
    display: inline;
  }
}
</style>
