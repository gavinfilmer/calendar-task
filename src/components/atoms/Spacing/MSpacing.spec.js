import { shallowMount, createLocalVue } from '@vue/test-utils'
import MSpacing from './MSpacing'

const localVue = createLocalVue()

describe('Spacing', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(MSpacing, {
      localVue,
      propsData: {
        top: 'sm',
        left: 'md',
        right: 'lg',
        bottom: 'xl',
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
