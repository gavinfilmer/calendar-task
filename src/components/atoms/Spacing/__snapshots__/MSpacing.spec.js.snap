// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing renders correctly 1`] = `
<div
  class="m-spacing m-spacing--top-sm m-spacing--right-lg m-spacing--bottom-xl m-spacing--left-md"
/>
`;
